### PR TITLE
Clarify rules regarding the timeline to removal of deprecated versions

### PR DESCRIPTION
**Pull request by mtaufen**
/cc @thockin

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/website/6034)

### CODE REVIEW

File: docs/reference/deprecation-policy.md [hunk 1]
```diff
@@ -64,11 +64,15 @@ knows how to convert between them in both directions.  Additionally, any new
 field added in v2 must be able to round-trip to v1 and back, which means v1
 might have to add an equivalent field or represent it as an annotation.
 
-**Rule #3: An API version in a given track may not be deprecated until a new
-API version at least as stable is released.**
+**Rule #3: An API version in a given track may not be removed until a new
+API version at least as stable has been available for at least the duration described in Rule #4a *and* the old API version has been announced deprecated for at least the duration described in Rule #4a.**
```
Comment (thockin) [inline docs/reference/deprecation-policy.md]: This feel like a no-op change.  Can you cite an example of what you are trying to allow/disallow with this change?

Comment (mtaufen) [reply docs/reference/deprecation-policy.md]: With this PR, I'm just trying to clarify the rules. 

The word "deprecated" is ill-defined. Some interpret it as "announcement of deprecation," some interpret it as "removal of the api version." The original Rule 3 is using the announcement interpretation, but some may read it as "I can delete v1 in the same release that v2 goes GA." I think the new wording makes it more clear that *removal* (the dangerous action) is guarded by the deprecation timeline. 

I also wanted to make it clear that the deprecation timeline can only start when both an announcement has been made and a migration option has become available. Rule 4a only mentions announcement. Though you could reasonably expect someone to assume that Rule 4a applies to both announcement and availability of a replacement, it pays to be explicit.

Comment (thockin) [reply docs/reference/deprecation-policy.md]: I think this makes it appear much more tricky.  I thought I was pretty careful to always use "deprecate (v)" as "to announce the deprecation" and "remove (v)" as "to remove the support code".

Would it be better to maybe just add words to that effect?

Comment (thockin) [reply docs/reference/deprecation-policy.md]: (e.g. at the beginning)

Comment (mtaufen) [reply docs/reference/deprecation-policy.md]: I think we should just be explicit and always say "announce" or "remove."

Comment (thockin) [reply docs/reference/deprecation-policy.md]: I thought I did that.   Do you want to re-tool this PR with your concrete changes?

**Comment by steveperry-53** [comment]
@mtaufen @thockin, I agree with @thockin on this one. For sure, we should give clear definitions of "deprecate" and "remove". But then we should use them as defined. To deprecate an element means one thing. To announce that deprecation (say, in a blog post) means something else. Having said that, I'm happy to go with whatever the two of you decide.

**Comment by thockin** [comment]
I'm closing this - if you want to discuss clarifications or enhancing precision, let's do that.

**Comment by mtaufen** [comment]
I don't have time to think about how to better word this until after code freeze, but I'll throw it on my todo list.

**Pull request closed without merging**